src/withForm/index.js

```diff
@@ -120,73 +49,28 @@ export const withForm = ({
         setErrors: state => value => ({ ...state, [formErrors]: value })
       }
     ),
-    withHandlers({
-      validateFormData: ({ setErrors, ...props }) => optionalData => {
-        const errors = validator(
-          optionalData || props[formData],
-          props[schema]
-        );
-        setErrors(errors);
-        return errors;
-      }
+    withHandlers(initialProps => {
+      // Expensive ops.  Do them once, keep in closure.
+      // API dict creation should really only happen once, or maybe on build.
+      const apiSchema = createApiDict(initialProps.apiSchema);
+      const validator = createValidator({ apiSchema, inputType: input });
+
+      return {
+        validateFormData: ({ setErrors }) => data => {
+          try {
```

Comment:
whitespace is your friend lol

Reply:
There are only four places prettier would let me, and none of them are helpful.

![image](https://user-images.githubusercontent.com/2559043/57556777-94e8e100-733d-11e9-939f-5625266a180a.png)

Reply:
Five.  Whatever, blame prettier, I like whitespace.